### PR TITLE
Update to Julia 1.4 and CUDA.jl

### PR DESCRIPTION
**Pull request by ali-ramadhan**
Need to update README and docs to mention the new Julia 1.4+ requirement.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -4,10 +4,7 @@ version = "0.30.0"
 
 [deps]
 Adapt = "79e6a3ab-5dfb-504d-930d-738a2a938a0e"
-CUDAapi = "3895d2a7-ec45-59b8-82bb-cfc6a382f9b3"
-CUDAdrv = "c5f51814-7f29-56b8-a69c-e4d8f6be1fde"
-CUDAnative = "be33ccc6-a3ff-5ff2-a52e-74243cff1e17"
-CuArrays = "3a865a2d-5b23-5a0f-bc46-62713ec82fae"
+CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
```
Comment (glwagner) [inline Project.toml:7]: 🦾

**Comment by glwagner** [comment]
Looks like `CUDA` exports a function named `device`, which conflicts with the definition in `.Architectures`.

**Comment by glwagner** [comment]
@ali-ramadhan tests pass on the CPU! The GPU error is cryptic --- can you take a look?

**Comment by ali-ramadhan** [comment]
Updated our `.gitlab-ci.yml` to look like https://github.com/JuliaGPU/CUDAnative.jl/blob/master/.gitlab-ci.yml

Looks like all tests pass now!

**Pull request merged**